base/cvd/allocd/alloc_utils.cpp

```diff
@@ -17,17 +17,19 @@
 
 #include <cstdint>
 #include <fstream>
+#include <string_view>
 
 #include "android-base/logging.h"
+#include "absl/strings/str_format.h"
 
 #include "cuttlefish/host/commands/cvd/utils/common.h"
 
 namespace cuttlefish {
 
-int RunExternalCommand(const std::string& command) {
+int RunExternalCommand(std::string_view command) {
   FILE* fp;
   LOG(INFO) << "Running external command: " << command;
-  fp = popen(command.c_str(), "r");
+  fp = popen(command.data(), "r");
```

Comment:
This is not safe. `std::string_view` does not guarantee null termination. If the argument represents a fragment of a larger `std::string`, `popen` will consume the rest of the string up until the null terminator.

If this needs to accept a `std::string_view`, it has to create a `std::string` copy so that it can guarantee there is a null terminator in the right place. As far as I can tell most of the callers create a `std::stringstream` and produce a string with `.str()`, so it seems more appropriate to still accept a `const std::string&`.